### ISSUE TITLE
Generated table of contents does not reflect on header order if it is a number

### ISSUE THREAD

**Issue by nicholascostadev** [vnphanquang/svelte-put#269]
I don't know if it is stated in the docs, but I'm having an issue where if the header is an integer, it gets rendered first in the list. I'm guessing that it's happening because objects are not always kept in order of insertion, so the final `$tocStore.items` is not follwing the correct order.

I created a sveltelab project for you to easily see this: [sveltelab link](https://www.sveltelab.dev/?files=.%2Fsrc%2Froutes%2F%2Bpage.svelte#code=N4IgdA5glgLlEDsD2AnApiAXKAZlANhtiAMZIIxoUDOWoA1lAgCZYjUBua%2Bl%2BAhgCMAtAFcoADj4gANCAEicONCiwBtACwA2aZvHTxAZmkBOAKz6jARkvbrl6dfUOADPcvOH7z2%2Bcf39sxdjbxc-SwB2Fz1-T3MvdUjjaIiXOOdo3xdpBOzbSzjLDJ9k21dstMj3AtLsyssAJhNE4PdEyucrMvjSty9rDNyXXrC9LWzGrwMjMZjrEcHZ%2Bo9jdpaGqKGs-tTPHsHjNeGXCec06dK88xmbHfcWrvzdpqDPRvUrk88M6s2xg5C5sdbh0FjdrFc8qU0vdej8Vi9rG8Pi4ALoAXwxsjAAAd0DA4MoUCQ6CA8IQSWQKFQYLRiIwWGxONxeIJRBIpLJ5IplGoGp0TNIDE5rHU0vZxE54UlYoKnKY9AYJmDLMKUjF1MLlrKZVVPH4Qe4ivpenVenshdJ5YKlXlVe17BqssELdZbc9dYbbn0QZpTQjnMF4RarYrtYC7HrNhbHV5naqlZqCmFtOJ2mlky4nMGFW9gupRqNNQKXetPSKXJ0kzLs9aTPYSwnNq6nh6Hk53q8NuX3JXIx3wV8m%2BkW2ULcZpkW452y2qDadI4ZNvDHCZ7k5dF7VeYa6HjPX29oxxOnWHrIPZirPInQk3HjvGnuHPUjKGXc4TmkjQP79r1G9LueXTDs2MTRlmjQlr0Npekc35yjmT4vhB8bupeA4Xlm8G1iWBRfsBYKjuBT7bhM9TmOOWxkaiGJorI2J8CQ9B8BAaBgAAVtQ5AkmSRCgJSlA0CS9KsJg7BcDw3CsmIkgyHIChKComCqHyJ4lssqxRlhr6Snoy6PLq-xgapqpfmh%2BnzhGgI-tGozaPmgzqFOMZag2V5erpdTmlp0Ert2ZTOcWW6rm5Hqlsq27eYhJ5Tu4%2Bq9AqkUlt8I71iccWDkZsaBSYyRpl2mTWSZXrmg%2BuXAhkHiZVqMVhZq8XAr0U4hshgE-KBSqNheyUXk5k7ZWWFl9DchXTmhDyQg1bmhh2-alvh7S9oCs1iqhkpruUN6hURYzTVc5HrTE-ytLWO3tn1MXvgCY2Bp52rNaenXjSOK0rrtG2Il2s6LfEBQrXpa2Zu9mRbSdh5Iu9lhrNeIGjv%2BYZag0Vx9W%2B6XbBuF4nLBwMPMs9wvXekUqVlU6aEF2yzPhezuABOTLpkek%2BW2G16bdEYjcYcoOnZtnZAl2TndqpM6oUQ4lEONPtHlzbs3KHhjPZ8t8wFgaC2TIsU2LFMS61KUQycZU9l6cu-YBxS6zLlpc7zDlKwLrm%2BU9MP45hloIWOsuDKYdnc7%2BdsoQ7ZrPbe7YFPrngLUbEO4aL5uJZKHvy3mPOOb70WntHGux672Hx5acMK9opgvmdadJcVmnZ9pJhyuc1uK-DAqI-zxktvUUEV-db6HJts5GlV-XY%2B3F76qjsZ47eEWV6R-IqxaQvoerQGa-5ofh08OQgWehlx9Xec2-v-ez2reHLw6%2B2A4d9yRBbphW6YcOmC7WVN30VHuOimLsESJLMFA6AkDAVAABPEkfBsTYjAAACxgAAW3wNxAgvFSDkAEjSISTARJiWZJJYQ0kORyW5IpVQmhKpGA3OEYIvp9BOHEMEcQHhNDOhOKqUo9w9CaE%2BCQi%2BBE2G1k9HpfUZD9y3kquuT4lD9R1XdJkDhacuF3FvNoIMD51r-R1E9TQwjuwEXBl7bU01IiyOJjofU3VvTX0embTR8Nz6WUmERdwqopHuEaNYi0BhRSm0%2BlTZwUsbh-E8ehR2d5Igb0unpZKR0zxLTyEqQR%2Bi3iGPEQKeRAZFHPFDICOmMIdBaMXvjNYSpegsIScFOxGZh5PDcaqcesURzrh8KPSJOTZq1OBgRF2BidDJOdIExeZsYY%2BD8d7C%2BZQ6aVXaCkuW7QpHjO6SeLh0pLLAzbnw%2Bmt14TLS8TOHxwy9YhSXiEOZ-Z1lPHhNU7h9UMKCx6SYzag0EIL1sKsga9gLl1NCqvfhlRVn3SibBAiAKAJzwgn0k%2ButjqhRGUs6JKz6yGOmfc6ORitRcNpskRpT43mcMRY4kcvCUUf1oiAfAUABA-z-mgABwCSQAGEkAiCpCgMATIJIIPJMQfi1JaQMAwYycSLJcHslklyBSah5GPCidDMK2gUWN2Ak9Xc3dtEZlDG4t4ehTD3FUoXR5FlLrTVxldOoqrXGuNStkJO2QLVE2qtI0yJT1DLCUcqtUQ0wZehIjeARFZg7GvUdEVZmTnDQznFmIwxg7L2C1Wnf5a9wzDXNbWUwkQY3PzfvqeVSoTVlIIgUI4wbJQ5o%2BQqt57RmkuFCXLG1z5Y2uuLYm7IQiTrBDTQjDNdyN6Sqhl6aCcqvCLINsqPIBaimRitEdBVtjfG5Kinin1JajRUIGtOn55g5WUPLTsSd9obyZNrXGlVrc0iNwPfWyM0sk3WPVZa2smgkKCjWFCf09YmFT0fUHBRF5twtA7WixIQ6JrlPme-GisgMFoAAB7sR5aSRBFIUHcvQQyUSrLBVshkpyeSPIlI5ByJkuaHkymwyBK2UyXqGi9BNHwyGqFTUtjPMG6Eo1ilPAI6UJx-rDZ4vY6Mi1kbBSlEibpD1dNImeKY5GXjeLPlrLHQNUeB5qI0WJSgBllBYO-3-oAlAIDiAAGp%2BBAPUyygVSCeIIapIJOkfLUNmf4EKzDBCxVKQlbYsaXybgbuBIUq6JTXx5jshHGcAQiOej%2BGsE2BhW3av-TKKVvaqmfAHWET8V0PU5EJWBkA%2Bn6IsRZcoLgzKOLsqQVy6zvKUNYIkg5jD%2BDRU4eUl0JYLGQqNsPUNH4waMXSMqkm3cgSXljxOtWu92K%2BG1ufmNQO2jVR9bnXarJGkYid0oUowJ85DKNAlEOXz3zzbat-eYNtxFLT3CJbIXLzFWJofM-BzliGKsgGEvy7BtW8Eiuw0QtzubpV9uSa2Pb0SzIepUtJreeYHHKgCLw0K%2BpQlEQS4dVmZQdsRn%2BIuTerH2qneDZ1WFb7os6GPjHQZ%2BztFBYYzDj93oBQddS2xnyEOLBU4ORam9SOxko-FKqanmPods%2ByKNlLtR4s9tCv9rYDCR7uRoxt6NmtmfWKJpN-SAuA7vqx8FfnI7BeOgI6F9R2P%2Bma%2BVHaKMpE4hUW86kj2si0cPDaoNeJ00JGi7xaYBhyTbeWi99Qien7vQPs952mZvuQOoskeH%2B3zi1fPvDCCAbvq6ZVjix7v3WUfch9kSH%2ByIfb6h8Btn73Ufi-h7z3LdhuKQ3R5xe77GyLkvS9QmkdMmxvNg9b0CUMFCcouD2FEkZ2wxPxsd0LwTrOMdWHV9NhYg0fGRetXP0y8SYbCg7a%2BCCW-sKuLCFuuVm%2Bd9H8P53pczRAY98DBkAfT7QRhfmgHp1E%2BtdT9ZxrmY8-dsynUPxiTTPqfhDT4L6naTDb6H7gFgHb6pLJQH6QHH5wEQF3qkIbQdgEYZyHJk4nJnAXrA5SJJ4pwpyUZAy1Jq5d4J7TDkSsxm7bqAYW4oHrqfCIFMEIFwFcJTD0FrKwj5RVjNgnL6gmxSgZBGizSowta8FpZw50ELzgqYF1D37dSZAdiHppIfRkZ0EzQMFZDMHwE6GH6LIWLxrmQK6T6w7O6Vqzoq4PpPJ%2BabD3Qv50JAE2HqhQ667-CAEQppL65jowrOxeJ2HUF%2BQ1pW5eqZaMEsG6GsHV4wFhERE6EqRhIBDn4yYb7hF6G4q9wR6xHxFSyXQvzN60wbZt4xBZafwQD4BIACB8D4BgAkDUCwYWYPZWZoI2ZVa3bvbCpYaEJqBWi%2BTA6Bz7ozwcFBKhLkREY7pAyRS56apyiHZDHmQjF95ShXwbQF4nJkzDj3QnZ5h7TqILHShLHmEzStp6rrGjCV7NzTQKgI4tp1xXFL5bFzEVDMxjHLEdj54nEBrj4KzbR3FeG3oKy-HRqzEaG7HPHPDjFvHGGFqfFP5TGWglyhiPx7weIXGNApqWiHihJAknggnDFgkHHHRvFokfFBJnHl7TGol7z3yCjXGImprwr3HAk7F4lKEvGHHHZGB2E16knj5wlIl-FwkokCk%2BxCnYlZS4nzEQxUxiH%2BJXDHFrKnFfGFgnS-HbStoMk-5nY4nMmSmzTSkZhvEeBcmKlP7fEqk0mUn57bhYlanik6lPF6lNgyl2THZQnMIwlGlGkaqWmYmWlomgqMnalRzsn1Jf7qhyma7cnmRkkpxFyUkEEFgWkCmckvg2kPESkOlO7OIqErzh7Gkel5nenClJlETUnUmim2lagZkhlr5hm5kF75k8mwnnHUl-EJnWxCmlmcxJlilVn2k1m5HA4ZirbSC97kxMwfR6RRaQ4nSUGjRhzhk6hDZO4%2BAzlqpCaRzHThBtDa5WDLkeHBDhBhBAqUlE6Y5ATT4%2BAWriDmDuFE5-HiDeE9Za5-FjliEvJzhHAPmjGGEVIpDKwpI7kJrraC4DTKkBajn8FuTKIs4Lp3hrkQSULJC87AjzaLhwU-qnmHmYyfSmTPr-AwXhCmS7r3lQ5zkJr9nXKtholMnBmElpDyburEFXTHnvpE5HmbgZKNDbksUhCkVrKuqsULwPhJFlAnZ5FZDzw5mYoGQGFP6qGz4uJ%2BoDhKHKr4SsZbKTrZk5JcnqQHI2kDEng9FmRUaM4LZ06PEhn7GiULGxkknRlKnxmqk3H2QVkPHbF0UzqsngmvHHZGn2VpamngXemdk3F6JuWzEeVbL0WLE2UrFuncKBXnFtnOWu7IlpmRWWUxXWXEYLFlkBUbBBVOUlnJqek9mVkWXVnZVskxVxmNkOVFUpUlWIkhUZVpxVVeUrlPgGkRn1VJV1w-HNVomhKAkVUqwdULFk4NA9V5kFXhbJWDWhWIkiltXGITVSlOkzUNlzXDiNWLW0lok1zlUPESXEzPlwoTaDGBhqXPmXiaUHRAQ6V6p6UOypqzGnUdqxjnV%2BBBrTzRQ1UzrcYmW-k3iqUAjqVuT%2BFhZ3XGyeV7EA15UOi9l043U2FQ2rRfGUWdUQrOkoF5jI33A3QHLfT%2BXmljCmAclf4nWDknknIiX%2BgSG-KPIixqKmlY2TU004FRpjU%2BXEa03uYDK4Gm7Q3tiw3RXY21kUwtIRmaniUZpEoqayDUAwB8BwDEjEBaZUo6Z6a4B8AcBQCUg4gIAQClaWaoKwYvZ2ZvZSQdHOaNbPgtCPiFB0IeBEWjluD8j1A1AeA%2B0OBGDuHhCuJ6CPnSC%2B1yzaBh2-iCih0x0eAO0OBO20L6DmDx2QxuBJ0NCRCR3sH0KjkpjrjBCR2p2J2UJOBx1E5p0OBJ1Czl2O3p2l0x3F313E6N0V0l06DmC10J3N013SBN1V2UKd191t091D393O0t1d2V0T290Wg7bj0Z1Z2x2eBPAh1u2LhT3t0L3BDb0t05C71poiy73H3cV92RDhD2CR2%2BaB36CL6%2B2Xi7331OC732C73qBWDP1L2%2B273uEbiaDmBo4tbH0j0D3D112gMv2RBv2BpgPd2gOPgCbAPgMT2P2wPT0Z2NBINwMoMgMT0F6rJYPoMtBGCEPt2WAkN6AH3cUvjT7r6IWnakPN2MOgO97MMoOSikSlAkO4NJ0EM8MtCapoNb38PR3rD1DB0SOeDT6YOu3aAv20Yv31CYMiNQMiMqRsPV3KPINJ1InGCNCb1MPHZj0iPiBvIGMQMiMaN5jTCcmqPaOL5OA31xnH2v2WOWM-LmMT2PBWNPjP1qN3XcP2NCNMNqN%2BNBO97v2ChlW9653TVRQ8U%2BMv3DiJOWOlhjmprHbcVuB90WQGCuNBPyMFBhPYO8NqPzi72MKWgp1qMcwOA8UrguMf2BMlM71qMfnNoDgxqKMUHBMWMFP%2BP5MtMhA%2BPSgU0QyRCAMeA3m9M4NBOTlaNDNJOtNzN8NBOakbhUQ%2BNEFv1yNuNBMh5E5tyVAok%2BOLhv3n5UQGDaA32707YGAmxJMNrFw6DNNENNozOlP9NfOLP3MOD9iH2v20b5BV4bQ%2BP3ofOF3L2u1kKHnsKUMiOaBKK7P2MK3Eq3Y1HkB4AQAwam2NHm3IaYJtE21OYNZEKA6ta9GibtAvUyp%2Bp45tarnYQN3SGk6Ar97f5AWnIKVUx3XkVuq65E7iWIp-jjB3olJEGqEka2rMYKXG52SVTMtq4yFsvUwcvmFQofqaljDLpDmkb6p7khGJJOiCXCyiYmtKWtgPo3q2ryp7J2GTZ2p4YGFzIMwQq8uoRUE2q6tli-XCx1AdROHhQ0b5ovqcE4FmIoUJ45LAXf5kVDg2pRlAZhTHajChKSzlzXDoGC17Af5NL16sVRJXCcs0uMwaV8uetyY2FJ7daerrw1DjVpu1juFKX7qmWzZhsJoRu8Vfo4xmWyYEZCWNoLz%2BsUsg5DjzZ2IOJDbUZoHpIwx4QTu6vVtKhfJjJtIfjBTBoApdvAY9u01xvWAkEuoX6MW3TSbckkYusjtTV3X1sChOu1jnlXkKkMaxIyhWgPuQqf6ZvbKUy-ClB5thIFsOPFuoSlttYUac2vQnCup4EbsJ6MWahE1Oy8VJ4wrUaXq9ZgdutBQf4WVXujQ8uRQEfct%2Bqmk%2BCnonpNxCubWjv%2BZpRDj01%2BRixKrdu5H6uVkXYgD62UAYsIBYs4vEANF8SPbNGVaEv2bEv1Zfa8j6pA5yuXXBu-unzquf77qQf46MsQXzxKuss%2BLaxfs-BdpmLmgxZS4ythQKeGXBqozsc8FlA6tYzweJ6W6Vu9Faf0PjldWCvarehuAymId9uDxAgszeJrK2fwd%2Bc6B%2BgIfAjTA-WDGSsGxzg8GPB-JcHjnZtDh7LkcmD8hNyy3nYqZAA)
![image](https://github.com/vnphanquang/svelte-put/assets/86835927/0a7fd014-34e7-46a7-9145-ef8401999894)

I guess by using a `Map`, it keeps the order of the inserted items, so it may be a solution for this issue, but I don't know what may be the performance implications of using it.

I'd like to know if it's a known issue or if there are any plans for it to be solved. I'm already taking a look, but I'm kinda lost since I never worked in a monorepo before

**Comment by nicholascostadev** [comment]
I managed to do it using a Map, but since it's not an object, the way to render the Toc, would be different, the example now looks like:

```svelte
{#if Object.values($tocStore.items).length}
    <ul>
      {#each Object.values($tocStore.items) as tocItem}
        <li>
          <!-- svelte-ignore a11y-missing-attribute -->
          <!-- eslint-disable-next-line svelte/valid-compile -->
          <a use:toclink={{ store: tocStore, tocItem, observe: true }} />
        </li>
      {/each}
    </ul>
  {/if}
```

Using the `Map`, would be like this:

```svelte
{#if $tocStore.items.size}
    <ul>
      {#each $tocStore.items.values() as tocItem}
        <li>
          <!-- svelte-ignore a11y-missing-attribute -->
          <!-- eslint-disable-next-line svelte/valid-compile -->
          <a use:toclink={{ store: tocStore, tocItem, observe: true }} />
        </li>
      {/each}
    </ul>
  {/if}
```

**Comment by vnphanquang** [comment]
Looks like `Map` is indeed the way to go. This will be released in the next major release. Thanks again

**Comment by nicholascostadev** [comment]
Thank you for reaching out, is there an intended date for the next major release?

**Comment by nicholascostadev** [comment]
That's ok, thank you for helping out!

**Comment by vnphanquang** [comment]
Available in [@svelte-put/toc@5.0.0](https://github.com/vnphanquang/svelte-put/releases/tag/%40svelte-put%2Ftoc%405.0.0). Thank you again for your help @nicholascostadev :tada:

**Comment by nicholascostadev** [comment]
You're welcome, you already helped me a lot at work with this library, it's an honor @vnphanquang!
